src/platform/testing/e2e/form-tester/form-filler.js

```diff
@@ -0,0 +1,266 @@
+const { parse: parseUrl } = require('url');
+const _ = require('lodash/fp');
+
+const FIELD_SELECTOR = 'input, select';
+const CONTINUE_BUTTON = '.form-progress-buttons .usa-button-primary';
+
+/**
+ * Finds the data in testData for a single field.
+ */
+const findData = (fieldSelector, testData) => {
+  const dataPath = fieldSelector
+    .replace(/^root_/, '')
+    .replace(/_/g, '.')
+    .replace(/\.(\d+)\./g, (match, number) => `[${number}]`);
+  const result = _.get(dataPath, testData);
+  return result;
+};
+
+const getElementSelector = (field, fieldData) => {
+  const inputSelector = `input[name="${field.selector}"]`;
+  const selectors = {
+    'select-one': `select[name="${field.selector}"]`,
+    checkbox: `input[id="${field.selector}"]${
+      fieldData ? ':not(checked)' : ':checked'
+    }`,
+    tel: inputSelector,
+    textarea: inputSelector,
+    text: inputSelector,
+    email: inputSelector,
+    number: inputSelector,
+    radio: `${inputSelector}[value="${
+      // Use 'Y' / 'N' because of the yesNo widget
+      // eslint-disable-next-line no-nested-ternary
```

Comment:
hah

Reply:
Yeah, I didn't like it, but I also didn't like pulling this out above the `selectors` declaration. If it's not clear, I can pull it up and not nest my ternaries.